git_theta/metadata.py

```diff
@@ -0,0 +1,167 @@
+"""Classes representing checkpoint metadata files"""
+
+import hashlib
+import dataclasses
+from collections import OrderedDict
+import re
+import json
+
+from git_theta import git_utils, utils
+from file_or_name import file_or_name
+
+
+class MetadataField:
+    def to_serializable(self):
+        d = dataclasses.asdict(self)
+        for k, v in d.items():
+            d[k] = v if v is None else str(v)
+        return d
+
+
+@dataclasses.dataclass(eq=True)
+class LfsMetadata(MetadataField):
+    version: str
+    oid: str
+    size: str
+
+    @classmethod
+    @property
+    def name(cls):
+        return "lfs_metadata"
+
+    @property
+    def lfs_pointer(self):
+        return (
+            f"version {self.version}\noid sha256:{self.oid}\nsize {self.size}".encode()
+        )
+
+    @classmethod
+    def from_pointer(cls, pointer_file):
+        match = re.match(
+            "^version (?P<version>[^\s]*)\s*oid sha256:(?P<oid>[^\s]*)\s*size (?P<size>[0-9]*)$",
+            pointer_file,
+        )
+        return cls(
+            version=match.group("version"),
+            oid=match.group("oid"),
+            size=match.group("size"),
+        )
+
+    @classmethod
+    def from_bytes(cls, b):
+        return cls.from_pointer(git_utils.git_lfs_clean(b))
```

Comment:
`git_lfs_clean` is the code that actually takes the `b` bytes and sends them to git lfs right? And that results in the bytes getting put into the lfs store?

I think it is surprising for a big chunk of computation/communication to happen inside the creation method for the metadata object. It seems like it should be a cheap operation and it also makes it hard to tell where the writing happens. I think this is fine for now but we may want to change it if we refactor the updates a bit to make the git-lfs write more explicit.

Reply:
Agreed. One possibility is to use the `git lfs pointer` command that only constructs the pointer file without copying to `.git/lfs`. The command gives a little extra output that would need to be stripped away before passing to `from_pointer`.